spec/controllers/orders_controller_spec.rb

```diff
@@ -667,62 +667,78 @@ def deliver; end
-        end
+        context "for a non reservation-only instrument" do
+          let(:instrument) { FactoryGirl.create(:setup_instrument, :timer, facility: @authable, facility_account: @facility_account) }
 
-        it "assigns a price policy and cost" do
-          do_request
-          expect(@order_detail.reload.price_policy).not_to be_nil
-          expect(@order_detail.actual_cost).not_to be_nil
-        end
+          before { setup_past_reservation }
 
-        context "canceled" do
-          before :each do
-            @params[:order_status_id] = OrderStatus.canceled.first.id
+          it "sets the order_detail states to completed because it's in the past" do
+            do_request
+            expect(assigns[:order].order_details).to all(be_complete)
+          end
+
+          it "sets the order_detail fulfilment dates to the order time" do
             do_request
+            assigns[:order].order_details.all? do |od|
+              expect(od.fulfilled_at).not_to be_nil
+              expect(od.fulfilled_at).to match_date reservation.reserve_end_at
+            end
           end
 
-          it "is able to set order_detail states to canceled" do
-            assigns[:order].order_details.all? { |od| expect(od.state).to eq("canceled") }
+          it "sets the actual times to the completed reservation times" do
+            do_request
+            expect(reservation.reload.actual_start_at).to match_date reservation.reserve_start_at
+            expect(reservation.actual_end_at).to match_date(reservation.reserve_start_at + 60.minutes)
           end
 
-          it "sets the canceled time on the reservation" do
-            assigns[:order].order_details.all? { |od| expect(od.reservation.canceled_at).not_to be_nil }
-            expect(@reservation.reload.canceled_at).not_to be_nil
-            # Should this match the date put in the form, or the date when the action took place
-            # @reservation.canceled_at.should match_date @submitted_date
+          it "assigns a price policy and cost" do
+            do_request
+            expect(@order_detail.reload.price_policy).not_to be_nil
+            expect(@order_detail.actual_cost).not_to be_nil
+          end
+
+          context "canceled" do
+            before :each do
+              @params[:order_status_id] = OrderStatus.canceled.first.id
+              do_request
+            end
+
+            it "is able to set order_detail states to canceled" do
+              expect(assigns[:order].order_details).to all(be_canceled)
+            end
+
+            it "sets the canceled time on the reservation" do
+              assigns[:order].order_details.all? { |od| expect(od.reservation.canceled_at).not_to be_nil }
+              expect(reservation.reload.canceled_at).not_to be_nil
+              # Should this match the date put in the form, or the date when the action took place
```

Comment:
I see this was already there, but we should either fix this spec or remove the comment.